cypress/tests/upload-app-via-link.spec.ts

```diff
@@ -0,0 +1,368 @@
+    cy.intercept(API_ROUTES.sbomProjectList.route).as('sbomProjectList');
+
+    cy.intercept('POST', API_ROUTES.uploadAppViaLink.route).as(
+      'uploadAppLinkReqPOST'
+    );
+  });
+
+  it(
+    'It successfuly uploads app via playstore link',
+    { retries: { runMode: 1, openMode: 1 } },
+    function () {
+      // Visit projects page
+      cy.visit(APPLICATION_ROUTES.projects);
+
+      // Necessary API call before showing dashboard elements
+      cy.wait('@submissionList', { timeout: 40000 });
+
+      cy.findByText(APP_TRANSLATIONS.startNewScan).should('exist');
+
+      uploadAppActions.closeUploadAppModalIfOpen();
+
+      cy.findByLabelText('uploadAppViaLink-btn')
+        .should('exist')
+        .should('not.have.attr', 'disabled');
+
+      cy.findByLabelText('uploadAppViaLink-btn').click();
+
+      cy.findByText(
+        APP_TRANSLATIONS.uploadAppModule.linkUploadPopupHeader
+      ).should('exist');
+
+      cy.findByPlaceholderText(
+        APP_TRANSLATIONS.uploadAppModule.linkPastePlaceholder
+      ).should('not.have.attr', 'disabled');
+
+      cy.findByPlaceholderText(
+        APP_TRANSLATIONS.uploadAppModule.linkPastePlaceholder
+      )
+        .type(playStoreUrl, { delay: 0 })
+        .then(($input) => {
+          expect($input.val()).to.equal(playStoreUrl);
+        });
+
+      cy.findByTestId('uploadAppViaLinkModal-confirmBtn').should(
+        'not.have.attr',
+        'disabled'
+      );
+      cy.findByTestId('uploadAppViaLinkModal-confirmBtn').click();
+
+      // Initiate link upload
+      cy.wait('@uploadAppLinkReq', { timeout: 60000 }).then(
+        ({ response: res }) => {
+          expect(res?.statusCode).to.equal(201);
+
+          uploadAppActions.openUploadAppModal();
+
+          cy.findByText(APP_TRANSLATIONS.viaLink, { timeout: 6000 }).should(
+            'exist'
+          );
+
+          cy.findByText(APP_TRANSLATIONS.uploadStatus).should('exist');
+
+          // Wait until "Validating URL..." text disappears and "Completed" text appears
+          cy.findByText('Validating URL...', { timeout: 90000 }).should(
+            'not.exist'
+          );
+        }
+      );
+
+      // Intercepts uploaded file submission request from uploaded file info
+      cy.wait('@uploadAppLinkReqPOST', { timeout: 60000 }).then(
+        ({ response: res }) => {
+          const uploadDetails =
+            res?.body as MirageFactoryDefProps['upload-app-link'];
+
+          const uploadedAppSubURL = `${API_ROUTES.submissionItem.route}/${uploadDetails.id}`;
+
+          // Intercept upload app submission request
+          cy.intercept('GET', uploadedAppSubURL).as('uploadedAppSubmission');
+        }
+      );
+
+      // Waits for upload app submission response
+      cy.wait(
+        [
+          '@uploadedAppSubmission',
+          '@uploadedAppSubmission',
+          '@uploadedAppSubmission',
+          '@uploadedAppSubmission',
```

Comment:
Do we have to wait for the fourth submission response?

Reply:
Yes, until the 4th response `submission.file` is undefined.